package/MDAnalysis/analysis/align.py

```diff
@@ -380,6 +377,175 @@ def alignto(mobile, reference, select="all", mass_weighted=False,
+        if os.path.exists(self.filename) and not force:
+            logger.warn("{0} already exists and will NOT be overwritten; use force=True if you want this".format(self.filename))
+
+        self.writer = _Writer(self.filename, **kwargs)
+        del _Writer
+        select = rms._process_selection(select)
+        ref_atoms = reference.select_atoms(*select['reference'])
+        traj_atoms = u.select_atoms(*select['mobile'])
+        natoms = traj_atoms.n_atoms
+
+        ref_atoms, traj_atoms = get_matching_atoms(ref_atoms, traj_atoms,
+                                        tol_mass=tol_mass, strict=strict)
+
+        logger.info("RMS-fitting on {0:d} atoms.".format(len(self.ref_atoms)))
+
+        if mass_weighted:
+            # if performing a mass-weighted alignment/rmsd calculation
+            weights = ref_atoms.masses
+        else:
+            weights = None
+
+        # reference centre of mass system
+        ref_com = ref_atoms.center_of_mass()
+        ref_coordinates = ref_atoms.positions - nref_com
+
+        # allocate the array for selection atom coords
+        traj_coordinates = traj_atoms.positions.copy()
+
+        nframes = len(frames)
+        self.rmsd = np.zeros((1,))
+
+
+        # R: rotation matrix that aligns r-r_com, x~-x~com
+        #    (x~: selected coordinates, x: all coordinates)
+        # Final transformed traj coordinates: x' = (x-x~_com)*R + ref_com
+        rot = np.zeros(9, dtype=np.float64)  # allocate space for calculation
+        R = np.matrix(rot.reshape(3, 3))
+
+        self.percentage = ProgressMeter(nframes, interval=10, quiet=quiet,
+                            format="Fitted frame %(step)5d/%(numsteps)d  [%(percentage)5.1f%%]\r")
+
+
+    def _single_frame(self):
+        x_com = traj_atoms.center_of_mass().astype(np.float32)
+        traj_coordinates[:] = traj_atoms.positions - x_com
+
+        # Need to transpose coordinates such that the coordinate array is
+        # 3xN instead of Nx3. Also qcp requires that the dtype be float64
+        # (I think we swapped the position of ref and traj in CalcRMSDRotationalMatrix
+        # so that R acts **to the left** and can be broadcasted; we're saving
+        # one transpose. [orbeckst])
+        np.append(self.rmsd, qcp.CalcRMSDRotationalMatrix(self.ref_coordinates.T.astype(np.float64),
+                    traj_coordinates.T.astype(np.float64), natoms, rot, weights))
+        R[:, :] = rot.reshape(3, 3)
+
+        # Transform each atom in the trajectory (use inplace ops to avoid copying arrays)
+        # (Marginally (~3%) faster than "ts.positions[:] = (ts.positions - x_com) * R + ref_com".)
+        self._ts.positions -= x_com
+        self._ts.frame.positions[:] = self._ts.positions * R  # R acts to the left & is broadcasted N times.
+        self._ts.frame.positions += ref_com
+
+        self.writer.write(u.atoms)  # write whole input trajectory system
+        self.percentage.echo(self._ts.frame)
+
+    def _conclude(self):
```

Comment:
Should we add in a `writer.close()` call here in case a writer has to add a footer?